spec/registry.md

```diff
@@ -255,3 +255,567 @@ which only expects the `accord` option:
+
+## String Value Selection and Formatting
+
+### Functions
+
+The following functions are provided:
+
+The function `:string` provides string selection and formatting.
+
+### Operands
+
+The _operand_ is any _literal_ value
+or an implementation-defined set of string or character-sequence types.
+The value of a _literal_ does not include any quotes around the _literal_ and 
+has converted any `quoted-escape` sequences to the escaped characters.
+It does include any enclosed spaces or control characters.
+
+> Examples of _literals_ and their values:
+> | Literal | Value | Notes |
+> |---|---|---|
+> | \|1\| | `1` | Does not include quotes `\|` |
+> | \|ab\\\|cd\| | `ab\|cd` | Unescape escaped `\|` |
+> | hello | `hello` | Unquoted literal |
+> | -13.2 | `-13.2` | Unquoted number-literal |
+> | \| spaces \| | ` spaces ` | Includes one space before and after |
+
+
+> [!NOTE]
+> This should probably include individual character types, such as `char`.
+
+In addition, implementations MAY perform formatting and selection on 
+`operand` values that do not otherwise have a formatting function registered.
+
+### Options
+
+The function `:string` has no options.
+
+> [!NOTE]
+> Proposals for string transformation options or implementation
+> experience with user requirements is desired during the Tech Preview.
+
+### Selection
+
+When implementing [`MatchSelectorKeys(rv, keys)`](spec/formatting.md#resolve-preferences), 
+the `:string` selector performs as described below.
+
+- Let `return_value` be a new empty list of strings.
+- Let `operand` be _rv_.
+If `operand` is not a string literal, let `operand` be
+the result of converting `operand` to a string literal.
+  or, optionally: emit a _Selection Error_ and return `return_value`.
+- Let `keys` be a list of strings containing keys to match.
+  (Hint: this list is an argument to `MatchSelectorKeys`)
+- For each string `key` in `keys`:
+   - If the value of `key` is equal to the string value of `operand`
+     then `key` matches the selector.
+     A `key` and an `operand` are equal if they consist of the same
+     sequence of Unicode code points.
+     Add `key` to the front of the `return_value` list.
+- Return `return_value`.
+
+> [!NOTE]
+> Matching of `key` and `operand` values is sensitive to the sequence of code points
+> in each string.
+> As a result, variations in how text can be encoded can affect the performance of matching.
+> The function `:string` does not perform case folding or Unicode Normalization of string values.
+> Users SHOULD encode _messages_ and their parts (such as _keys_ and _operands_),
+> in Unicode Normalization Form C (NFC) unless there is a very good reason
+> not to.
+> See also: [String Matching](https://www.w3.org/TR/charmod-norm)
+
+> [!NOTE]
+> Unquoted string literals in a _variant_ do not include spaces.
+> If users wish to match strings that include whitespace
+> (including U+3000 `IDEOGRAPHIC SPACE`)
+> to a key, the `key` needs to be quoted.
```

Comment:
Nitpick:
   s / needs to be quoted / MUST be quoted /
?

Reply:
This is a non-normative note, so using MUST would be out of place.